mlrun/db/httpdb.py

```diff
@@ -1833,21 +1833,31 @@ def create_project_secrets(
         ] = schemas.SecretProviderName.vault,
         secrets: dict = None,
     ):
-        """ Create needed Vault structures for storing secrets in project-context, and store a set of secret values.
-        The method generates Kubernetes service-account and the Vault authentication structures that are required
-        for function Pods to authenticate with Vault and be able to extract secret values passed as part of their
-        context.
+        """ Create project-context secrets using either ``vault`` or ``kubernetes`` provider.
+        When using with Vault, this will create needed Vault structures for storing secrets in project-context, and
+        store a set of secret values. The method generates Kubernetes service-account and the Vault authentication
+        structures that are required for function Pods to authenticate with Vault and be able to extract secret values
+        passed as part of their context.
 
         Note:
-                This method is currently in technical preview, and requires a HashiCorp Vault infrastructure
-                properly set up and connected to the MLRun API server.
+                This method used with Vault is currently in technical preview, and requires a HashiCorp Vault
+                infrastructure properly set up and connected to the MLRun API server.
+
+        When used with Kubernetes, this will make sure that the project-specific k8s secret is created, and will
+        populate it with the secrets provided, replacing their values if they exist.
```

Comment:
Just a matter of style (so since this is the only comment I'm merging anyway and if you want fix in a followup PR):
The paragraph of the explanation for "When using with Vault" is pretty long, and the word Kubernetes (which is also a provider itself) exists there, the combination of the two makes it's pretty confusing when you read the paragraph that you're still in the "when using with Vault" context, so I would suggest making it more of a title, and add indentation for anything "in it" e.g.:
```
Vault provider:
        text text text
Kubernetes provider:
        text text text
```

Reply:
Well, there isn't really an option to indent (or to create sections, for that matter) within docstrings coming from autodoc. I can organize it a little bit and manually create "headers" to separate the text into 2 parts, but the difference won't be significant IMO.